homeassistant/components/homematicip_cloud.py

```diff
@@ -106,13 +108,68 @@ def __init__(self, hass, config, websession):
         self.home = AsyncHome(hass.loop, websession)
         self.home.set_auth_token(_authtoken)
 
+        self.home.on_update(self.update)
+        self._accesspoint_connected = True
+
         hass.bus.async_listen_once(EVENT_HOMEASSISTANT_STOP, self.close())
 
     async def init(self):
         """Initialize connection."""
         await self.home.init(self._accesspoint)
         await self.home.get_current_state()
 
+    def update(self, *args, **kwargs):
+        """Update the home device.
+
+        Triggered when the hmip HOME_CHANGED event has fired.
+        There are several occasions for this event to happen.
+        We are only interested to check whether the access point
+        is still connected. If not, device state changes cannot
+        be forwarded to hass. So if access point is disconnected all devices
+        are set to unavailable.
+        """
+        if not self.home.connected:
+            _LOGGER.error(
+                "HMIP access point has lost connection with the cloud")
+            self._accesspoint_connected = False
+            self.set_all_to_unavailable()
+        else:
+            if not self._accesspoint_connected:
+                # Explicitly getting an update as device states might have
+                # changed during access point disconnect."""
+
+                job = self._hass.async_add_job(self.get_state())
```

Comment:
Is get_state async or sync method ? If async, any reason we would not make update async?

Reply:
`get_state` is async.
`update` is called from within the library when an update is received via the websocket connection. I can't make it async. Library does not support that. So I need to schedule.

is `async_add_job` the proper way to schedule a task or should I `hass.loop.create_task` like used in the example ?

https://developers.home-assistant.io/docs/en/asyncio_categorizing_functions.html#the-coroutine-function

Reply:
Use async_add_job. It is a wrapper around create_task.